index.html

```diff
@@ -122,6 +159,165 @@ <h2>Methods</h2>
       </dl>
     </section>
   </section>
+  <section id="rtcrtptransceiver-interface">
+    <h3>
+      <dfn>RTCRtpTransceiver</dfn> extensions
+    </h3>
+    <p>
+      The <code><a href="https://w3c.github.io/webrtc-pc/#dom-rtcrtptransceiver">RTCRtpTransceiver</a></code>
+      interface is defined in [[WEBRTC]]. This document extends that interface
+      by adding an additional method and attribute in order to control negotiation of
+      RTP header extensions.
+    </p>
+    <pre class="idl">
+partial dictionary RTCRtpHeaderExtensionCapability {
+  RTCRtpTransceiverDirection direction = "sendrecv";
+};
+      
+partial interface RTCRtpTransceiver {
+  void setOfferedRtpHeaderExtensions(
+    sequence&lt;RTCRtpHeaderExtensionCapability&gt; headerExtensionsToOffer);
+  readonly attribute FrozenArray&lt;RTCRtpHeaderExtensionCapability&gt; headerExtensionsNegotiated;
+};
+    </pre>
+    <p>
+      Let [[\HeaderExtensionsToOffer]] be an internal slot of the RTCRtpTransceiver,
+      initialized to the platform-specific initial set of RTP header extensions that
+      is offered on an initial CreateOffer. This list MUST include all header extensions
+      that are marked Mandatory to Implement in the underlying protocol. The direction
+      attribute for all extensions that are mandatory to use MUST be initialized to
+      "sendrecv".
+    </p>
+    <p class="note">The list of header extensions that MUST/SHOULD be supported is listed in
+      draft-ietf-rtcweb-rtp-usage, section 5.2. The "mid" extension is mandatory to use when
+      BUNDLE is in use.
+    </p>
+    <p>
+      Let [[\HeaderExtensionsNegotiated]] be an internal slot of the RTCRtpTransceiver,
+      initialized to an empty list.
+    </p>
+    <p>
+      In the algorithms for generating initial offers in JSEP section 5.1.1,
+      read "for each supported RTP header extension" as "For each RTP header extension
+      listed in [[\HeaderExtensionsToOffer]] where direction is not "stopped"".
```

Comment:
The "for each supported RTP header extension" language is in JSEP 5.3.1 (Constructing an Offer - Initial Offers), not 5.1.1.

We need to say how the direction affects the SDP, a spec reference is enough. E.g. "The direction attribute of the a=extmap (RFC5285, Section 6) MUST reflect the direction attribute of the RTCRtpHeaderExtensionCapability".

What about for the answer? You did not modify the paragraph in JSEP 5.3.1 (Generating an Answer - Initial Answers). Is this an offer-only API or is it meant to control what is answered as well? Can I use this API to reject an RTP header extension the same way RTCRtpTransceiverDirection is used in both the offer AND answer in the case of RTCRtpTransceivers? Or will the answer always reply with the maximum direction it is capable of receiving, regardless if the endpoint wants to offer it?